app/components/ChatHeader.tsx

```diff
@@ -2,19 +2,37 @@ import React from 'react';
 
 interface ChatHeaderProps {
   onClose: () => void;
+  onMaximize: () => void;
+  isMaximized: boolean;
 }
 
-const ChatHeader: React.FC<ChatHeaderProps> = ({ onClose }) => {
+const ChatHeader: React.FC<ChatHeaderProps> = ({ onClose, onMaximize, isMaximized }) => {
   return (
-    <div className="absolute top-2 right-0 z-50 pr-4">
+    <div className="sticky top-0 right-0 z-50 flex justify-end gap-2 bg-[var(--background)] px-4 py-2">
+      <button
+        onClick={onMaximize}
+        className="hidden md:inline-flex p-2 cursor-pointer hover:bg-gray-200 dark:hover:bg-gray-700 rounded w-8 h-8"
+        aria-label={isMaximized ? 'Minimize' : 'Maximize'}
+        aria-pressed={isMaximized}
+      >
+        {isMaximized ? (
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" className="lucide lucide-minimize-icon lucide-minimize">
```

Comment:
Let's not use `svg` here but importing the icon from `lucide-react` and use it as a component. Example: https://github.com/vinhlee95/stonkie_client/blob/main/app/components/Chat.tsx